internal/middleware/errorHandler.go

```diff
@@ -14,11 +19,17 @@ func ErrorHandler() gin.HandlerFunc {
 		if err == nil {
 			return
 		}
-		if c.Writer.Status() != http.StatusOK {
```

Comment:
This was used to not override status codes already set in a handler. I think we still want to respect that. A handler might still have a super specific case that we cannot/do not want to cover in this middleware. In that case it will have done `_ = c.AbortWithError(http.SuperSpecificStatus, err)` which, with your changes would be overridden 🤔